### ISSUE TITLE
Cannot create derived fields from particle filter types

### ISSUE THREAD

**Issue by jzuhone** [yt-project/yt#1474]
The following script:

```python
import yt

@yt.particle_filter(requires=["particle_type"], filtered_type='all')
def stars(pfilter, data):
    filter = data[(pfilter.filtered_type, "particle_type")] == 2
    return filter

ds = yt.load('IsolatedGalaxy/galaxy0030/galaxy0030')
ds.add_particle_filter('stars')

def _big_time(field, data):
    return 3.0*data["stars", "creation_time"]
ds.add_field(("stars", "big_time"), function=_big_time, units="Gyr", particle_type=True)

ds.all_data()["stars", "big_time"]
```

produces the following output

```pycon
---------------------------------------------------------------------------
YTFieldNotFound                           Traceback (most recent call last)
<ipython-input-5-a9608ddcb55a> in <module>()
----> 1 ds.all_data()["stars", "big_time"]

/Users/jzuhone/Source/yt/yt/data_objects/data_containers.py in __getitem__(self, key)
    279                 return self.field_data[f]
    280             else:
--> 281                 self.get_data(f)
    282         # fi.units is the unit expression string. We depend on the registry
    283         # hanging off the dataset to define this unit object.

/Users/jzuhone/Source/yt/yt/data_objects/data_containers.py in get_data(self, fields)
   1277             f = self.ds.known_filters[filter_type]
   1278             with f.apply(self):
-> 1279                 self.get_data(apply_fields[filter_type])
   1280         fields = nfields
   1281         if len(fields) == 0: return

/Users/jzuhone/Source/yt/yt/data_objects/data_containers.py in get_data(self, fields)
   1267         nfields = []
   1268         apply_fields = defaultdict(list)
-> 1269         for field in self._determine_fields(fields):
   1270             if field[0] in self.ds.filtered_particle_types:
   1271                 f = self.ds.known_filters[field[0]]

/Users/jzuhone/Source/yt/yt/data_objects/data_containers.py in _determine_fields(self, fields)
   1119                     raise YTFieldNotParseable(field)
   1120                 ftype, fname = field
-> 1121                 finfo = self.ds._get_field_info(ftype, fname)
   1122             elif isinstance(field, DerivedField):
   1123                 ftype, fname = field.name

/Users/jzuhone/Source/yt/yt/data_objects/static_output.py in _get_field_info(self, ftype, fname)
    749                     self._last_finfo = self.field_info[(ftype, fname)]
    750                     return self._last_finfo
--> 751         raise YTFieldNotFound((ftype, fname), self)
    752 
    753     def _setup_classes(self):

YTFieldNotFound: Could not find field '('all', 'big_time')' in galaxy0030.
```

I'm happy to look into this myself but might need a suggestion from someone who knows particle filters better as to why this may be happening.

**Comment by jzuhone** [comment]
Argh, sorry--I think I just noticed that this is the same as issue #1278. Right @ngoldbaum?

**Comment by ngoldbaum** [comment]
Does it work if you switch the order of `add_particle_filter` and `add_field`? If so, yes, it's the same issue.

We should probably make yt emit some kind of error or warning (or just stfu and do the right thing if we can).

**Comment by jzuhone** [comment]
It does work, but you have to use ``"all"`` as the field type in the field definition, which to my thinking isn't transparent:

```python
ds = yt.load('IsolatedGalaxy/galaxy0030/galaxy0030')
def _big_time(field, data):
    return 3.0*data["all", "creation_time"]
ds.add_field(("all", "big_time"), function=_big_time, units="Gyr", particle_type=True)
ds.add_particle_filter('stars')
```
I suppose it's not the worst thing, but you're right that we should either just fix it or the error message. I don't know enough about field detection to know for sure offhand what's happening, but I can dive in.

**Comment by ngoldbaum** [comment]
Feel free! We definitely need to fix it before 3.4.

Might want to look at this PR, the code paths I changed there might be pertinent: https://bitbucket.org/yt_analysis/yt/pull-requests/2569/fix-covering-grid-access-for-particle/diff#comment-34271330

Closing this issue in favor of #1278.
